index.js

```diff
@@ -16,12 +16,10 @@ module.exports = function (data, options) {
 			return;
 		}
 
-		if (file.data) {
-			data = _.merge(file.data, data);
-		}
+		var data = getData(file);
 
 		try {
-			file.contents = new Buffer(template(file.contents.toString(), data || {}, options));
+			file.contents = new Buffer(template(file.contents.toString(), data, options).toString());
```

Comment:
that is correct. if `data` is `null` the `template()` function will return a JS function, if `data` is anything else it will return the compiled template as a string. that means that we are calling either `String.toString()` or `Function.toString()` and since the former is a noop it shouldn't make a difference.